R/utils.R

```diff
@@ -327,6 +328,19 @@ check_list_of_bool <- function(x, names, arg = caller_arg(x), call = caller_env(
   }
 }
 
+with_indexed_errors <- function(expr,
+                                message,
+                                error_call = caller_env(),
+                                envir = caller_env()) {
+  try_fetch(
+    expr,
+    purrr_error_indexed = function(cnd) {
+      envir <- new_environment(list(cnd = cnd), parent = envir)
```

Comment:
It's surprising to use the same variable name. Should use a new one so it's immediately clear that we're not interpolating in `envir = caller_env()`.